bgflow/distribution/sampling/mcmc.py

```diff
@@ -1,35 +1,257 @@
+"""Markov-Chain Monte Carlo iterative sampling.
```

Comment:
This file now implements the more general Monte Carlo sampling stuff.
I am also leaving the old implementation in for backwards-compatibility. But I want to take it out as soon as we have a working TSF notebook.